test/unit/CMakeLists.txt

```diff
@@ -14,6 +14,12 @@ DLAF_addTest(test_types
   USE_MAIN PLAIN
 )
 
+DLAF_addTest(test_util_cuda
+  SOURCES test_util_cuda.cu
+  LIBRARIES dlaf.core
+  USE_MAIN PLAIN
+)
+
```

Comment:
Shouldn't this be guarded with `DLAF_WITH_CUDA` so that it is built just if CUDA is enabled?

I see there are also `__host__` functions tested there, but those are available just if `DLAF_WITH_CUDA=on`. Moreover, they are in a `.cu` file so a CUDA compiler is required.